kratos/solving_strategies/schemes/residual_based_adjoint_bossak_scheme.h

```diff
@@ -501,77 +846,139 @@ class ResidualBasedAdjointBossakScheme : public Scheme<TSparseSpace, TDenseSpace
                                       std::vector<const VariableData*>& rVec) {
                 return rExtensions.GetSecondDerivativesVariables(rVec);
             });
+        auto auxiliary_vars = GatherVariables(
+            rModelPart.Elements(), [](const AdjointExtensions& rExtensions,
+                                      std::vector<const VariableData*>& rVec) {
+                return rExtensions.GetAuxiliaryVariables(rVec);
+            });
+
         SetToZero_AdjointVars(lambda2_vars, rModelPart.Nodes());
         SetToZero_AdjointVars(lambda3_vars, rModelPart.Nodes());
 
-        const int number_of_elements = rModelPart.NumberOfElements();
-        const ProcessInfo& r_process_info = rModelPart.GetProcessInfo();
+        const auto& r_process_info = rModelPart.GetProcessInfo();
+        UpdateEntityTimeSchemeContributions(rModelPart.Elements(), r_process_info);
+        UpdateEntityTimeSchemeContributions(rModelPart.Conditions(), r_process_info);
+
+        // Finalize global assembly
+        Assemble_AdjointVars(lambda2_vars, rModelPart.GetCommunicator());
+        Assemble_AdjointVars(lambda3_vars, rModelPart.GetCommunicator());
```

Comment:
what does this do?

I am really not clear about the type of the variables names `lambda`. Are those lambdas or not?

Reply:
In adjoint we use lambda as in greek alphabet lambda to store the solution (Those are solution variables). Those two methods will assemble all the variables (from `VariableData` ) in the model part.

Reply:
ahhh
that is quite confusing for someone not familiar with this 

up to you to change it or not

Reply:
the usage of ``lambda`` is quite common in adjoint sensitivity analysis.

Reply:
maybe a good middle ground would be to specify the type instead of using `auto`

Reply:
done :)